tensorpipe/test/channel/channel_test.h

```diff
@@ -202,6 +203,62 @@ class ChannelTestCase {
 
 template <typename TBuffer>
 class ClientServerChannelTestCase : public ChannelTestCase<TBuffer> {
+  std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
+  accept(
+      std::shared_ptr<tensorpipe::transport::Listener> listener,
+      size_t numConnections) {
+    auto promises = std::make_shared<std::vector<
+        std::promise<std::shared_ptr<tensorpipe::transport::Connection>>>>(
+        numConnections);
+    for (size_t i = 0; i < numConnections; ++i) {
+      listener->accept(
+          [&, promises](
+              const tensorpipe::Error& error,
+              std::shared_ptr<tensorpipe::transport::Connection> connection) {
+            ASSERT_FALSE(error) << error.what();
+            connection->read([promises, connection](
+                                 const tensorpipe::Error& error,
+                                 const void* connIdBuf,
+                                 size_t length) {
+              ASSERT_FALSE(error) << error.what();
+              ASSERT_EQ(sizeof(uint64_t), length);
+              promises->at(*static_cast<const uint64_t*>(connIdBuf))
+                  .set_value(std::move(connection));
+            });
+          });
+    }
+
+    std::vector<std::future<std::shared_ptr<tensorpipe::transport::Connection>>>
+        futures;
+
+    for (auto& p : *promises) {
+      futures.push_back(p.get_future());
+    }
+
+    return futures;
+  }
+
+  std::vector<std::shared_ptr<tensorpipe::transport::Connection>> connect(
+      std::shared_ptr<tensorpipe::transport::Context> transportCtx,
+      std::string addr,
+      size_t numConnections) {
+    std::vector<std::shared_ptr<tensorpipe::transport::Connection>> connections(
+        numConnections);
+    for (size_t connId = 0; connId < numConnections; ++connId) {
+      connections[connId] = transportCtx->connect(addr);
+      auto connIdBuf = std::make_shared<uint64_t>();
+      *connIdBuf = connId;
```

Comment:
```suggestion
      auto connIdBuf = std::make_shared<uint64_t>(connId);
```
Wondering if this would work?